src/grt/src/heatMap.cpp

```diff
@@ -144,8 +144,12 @@ bool RoutingCongestionDataSource::populateMap()
     return false;
   }
 
-  auto gcell_congestion_data = grid->getCongestionMap(layer_);
-  if (gcell_congestion_data.numElems() == 0) {
+  auto gcell_hor_congestion_data
+      = grid->getCongestionMap(layer_, odb::dbTechLayerDir::HORIZONTAL);
+  auto gcell_ver_congestion_data
+      = grid->getCongestionMap(layer_, odb::dbTechLayerDir::VERTICAL);
```

Comment:
Not always. When we use the Direction == ALL, both hor and ver data will be filled with values. When using a specific direction, the other one will be zero.

Reply:
In dbGCellGrid::getCongestionMap 
```
if (direction == tech_layer->getDirection()) {
```
will block the use of ALL unless the layer itself has that direction.  Is that what you are describing?  It would be unusual to route on such a layer.

Reply:
Sorry, I made a mistake in my previous comment. Both congestion data will have values when we want to show the congestion in all layers. In this case, we'll get the hor and ver data from all layers, and each of the congestion data vars will be filled with the correct amount of capacity and usage of the specified direction.

When we choose to show only one layer, one of the congestion data will be filled with zero, as you mentioned.